components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,13 +164,92 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
 	}
 
-	if err := oAx.home(ctx, np); err != nil {
+	if err := oAx.home(context.Background(), np); err != nil {
 		return nil, err
 	}
 
+	oAx.checkHit()
 	return oAx, nil
 }
 
+func (g *oneAxis) checkHit() {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	g.cancelFunc = cancelFunc
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return nil
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
```

Comment:
talked to rand and I'm going to make another ticket to switch the implementation to digital interrupts later, and just use the busy loop for now!

Reply:
I'm content with that, but please put a comment into the code saying something like 
```go
// TODO(RSDK-1234) Make this asynchronous using digital interrupts to avoid the busy loop.
```